### DISCUSSION TITLE
Babe's run_one_test() doesn't initialize blocks?

### DISCUSSION THREAD

**Discussion by nazar-pc** [paritytech/substrate#9434] [Q&A] [answered]
Debugging another issue with code based on BABE and can't understand why `run_one_test()` doesn't seem to initialize blocks: https://github.com/paritytech/substrate/blob/7dcc77b982f59eaf6cec19499d981164b04a255d/client/consensus/babe/src/tests.rs#L387-L492

What I mean by that is that `on_initialize` hook isn't called in my pallets and when I call other methods and store data, I can't retrieve them with runtime API later.

Maybe `substrate-test-runtime` is configured in such a way that hooks are not executed and I need to tweak it somehow?

**Comment by xlc** [comment]
Maybe related https://github.com/paritytech/substrate/pull/8953

**Reply by nazar-pc** [reply]
I forked substrate on April 11th, so it doesn't include that change yet.

**Comment by bkchr** [comment, marked as answer]
`substrate-test-runtime` is not using FRAME. You should read the code and you would see it.
